### DISCUSSION TITLE
TraceQueryEsDAO query segments size limit

### DISCUSSION THREAD

**Discussion by dogblues** [apache/skywalking#11531] [General]
When query single trace By traceId, **TraceQueryEsDAO** has a query segments size limit (default value 200):

```Java
final SearchBuilder search =
            Search.builder()
                  .query(Query.term(SegmentRecord.TRACE_ID, traceId))
                  .size(segmentQueryMaxSize);
```
Meanwhile in **JDBCTraceQueryDAO** and **BanyanDBTraceQueryDAO**, there are no similar limits. In produce environment, system is complicate so a single trace could have a huge mount of segments. The segmentQueryMaxSize limit would result in incomplete trace struct. 

Why doesn't  TraceQueryEsDAO use **scrolling search**  to query all trace segments, which is used in **MetadataQueryEsDAO#listServices** etc.

**Comment by wu-sheng** [comment]
I think it is missed to change.
I don't think there is a specific reason rather than on purpose.

**Reply by wu-sheng** [reply]
But rather than 200 segments seem unreasonable.

**Reply by dogblues** [reply]
If it's reasonble, I can submit a PR to do it.

**Reply by wu-sheng** [reply]
How does a trace ID have 200+ segments?

**Reply by dogblues** [reply]
multi-threading execute / MQ topic many consumers

**Reply by wu-sheng** [reply]
But this is only one trace, means, one queue message, or one time bulk.
Are they over 200?
I have doubts whether UI and browser could live?

**Reply by dogblues** [reply]
Yes, in our cases, one trace may have thousands of segmens.
Your doubt makes sense. We have tried modify the limit config to query all segments, UI would be very slow or even freeze

**Reply by wu-sheng** [reply]
That is my feeling. There are too many.

**Reply by wu-sheng** [reply]
I think fixing the query doesn't resolve your issue actually.
